lib/elixir_boilerplate_graphql/operation_name_logger.ex

```diff
@@ -0,0 +1,14 @@
+defmodule ElixirBoilerplateGraphQL.OperationNameLogger do
+  @behaviour Absinthe.Middleware
+
+  def call(resolution, _opts) do
+    with %Absinthe.Blueprint.Document.Operation{name: name} <- Enum.find(resolution.path, &current_operation?/1) do
+      Logger.metadata(graphql_operation_name: name)
+    end
+
+    resolution
+  end
+
+  defp current_operation?(%Absinthe.Blueprint.Document.Operation{current: true}), do: true
+  defp current_operation?(_), do: false
```

Comment:
I would also add an `alias` like @simonprev mentioned in the other thread.

```suggestion
  @behaviour Absinthe.Middleware
  
  alias Absinthe.Blueprint.Document.Operation

  def call(resolution, _opts) do
    with %Operation{name: name} <- Enum.find(resolution.path, &current_operation?/1) do
      Logger.metadata(graphql_operation_name: name)
    end

    resolution
  end

  defp current_operation?(%Operation{current: true}), do: true
  defp current_operation?(_), do: false
```